### PR TITLE
feat: add address field

### PR DESCRIPTION
**Pull request by lucachaco**
image

### CODE REVIEW

**Comment by lucachaco** [comment]
Done

File: src/stories/AddressField.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+import { ComponentStory, ComponentMeta } from '@storybook/react'
+
+import themeDecorator from './decorators'
+import AddressField from '../components/AddressField'
+
+export default {
+  title: 'Components/TextField',
```
Comment (chescalante) [inline src/stories/AddressField.stories.tsx:8]:
```suggestion
  title: 'Components/AddressField',
```

File: src/stories/AddressField.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    }
+  },
+  decorators: [themeDecorator]
+} as ComponentMeta<typeof AddressField>
+
+const Template: ComponentStory<typeof AddressField> = (args) => <AddressField {...args} />
+
+export const Primary = Template.bind({})
+Primary.args = {
+  children: 'TextBu',
```
Comment (chescalante) [inline src/stories/AddressField.stories.tsx:22]:
```suggestion
```

**Review by chescalante — CHANGES_REQUESTED** [review]

File: src/components/AddressField.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
+
+import React, { useState } from 'react'
+import { TextField } from '@mui/material'
+import Resolver from '@rsksmart/rns-resolver.js'
+
+// @ts-ignore
+// eslint-disable-next-line new-cap
+const resolver = new Resolver.forRskTestnet()
+
+const AddressField = ({ label, onReceiverAddressChange } : any) => {
```
[the remark this answers is not in the document]
Comment (lucachaco) [reply src/components/AddressField.tsx]: Done

File: src/components/AddressField.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  const [to, setTo] = useState('')
+  const [resolverStatus, setResolverStatus] = useState('')
+  const rskDomainValidation = (e:any) => {
```
[the remark this answers is not in the document]
Comment (lucachaco) [reply src/components/AddressField.tsx]: Done

File: src/stories/AddressField.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  label: '',
+  onReceiverAddressChange: () => {}
+}
+
+export const Secondary = Template.bind({})
+Secondary.args = {
+  color: 'secondary',
+  label: '',
+  onReceiverAddressChange: () => {}
+}
```
[the remark this answers is not in the document]
Comment (lucachaco) [reply src/stories/AddressField.stories.tsx]: Done

File: src/stories/AddressField.stories.tsx (hunk 1 above)
Comment (lucachaco) [reply src/stories/AddressField.stories.tsx]: Done

File: src/stories/AddressField.stories.tsx (hunk 2 above)
Comment (lucachaco) [reply src/stories/AddressField.stories.tsx]: Done

**Comment by lucachaco** [comment]
Last LGTM comment also fixed

File: src/components/AddressField.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  const [resolverStatus, setResolverStatus] = useState('')
+  const rskDomainValidation = (e:IEvent) => {
+    const selectedTo = (e && e.target && e.target.value)
+    setTo(selectedTo)
+    if (!selectedTo) {
+      return
+    }
+
+    const re = /\.rsk$/ // match *.rsk domains
+    const isValid = re.test(String(selectedTo).toLowerCase())
```
Comment (ilanolkies) [inline src/components/AddressField.tsx:27]:
```suggestion
    const isDomain = re.test(String(selectedTo).toLowerCase())
```

File: src/components/AddressField.tsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  target: { value }
+}
+export interface IAddressFieldInput{
+  label: string
+  onReceiverAddressChange: (address: string) => void
+}
+const AddressField = ({ label, onReceiverAddressChange } : IAddressFieldInput) => {
+  const [to, setTo] = useState('')
+  const [resolverStatus, setResolverStatus] = useState('')
+  const rskDomainValidation = (e:IEvent) => {
```
Comment (ilanolkies) [inline src/components/AddressField.tsx:19]:
```suggestion
  const setAddressOrDomainResolution = (e:IEvent) => {
```

**Review by ilanolkies — CHANGES_REQUESTED** [review]
LRGTM!

File: src/components/AddressField.tsx [hunk 8]
```diff
@@ -0,0 +1,56 @@
+
+import React, { useState } from 'react'
+import { TextField } from '@mui/material'
+import Resolver from '@rsksmart/rns-resolver.js'
+
+// @ts-ignore
+// eslint-disable-next-line new-cap
+const resolver = new Resolver.forRskTestnet()
```
Comment (ilanolkies) [inline src/components/AddressField.tsx:8]: We should pass this as parameter of the component

File: src/components/AddressField.tsx (hunk 7 above)
Comment (lucachaco) [reply src/components/AddressField.tsx]: Done

File: src/components/AddressField.tsx (hunk 6 above)
Comment (lucachaco) [reply src/components/AddressField.tsx]: Done

File: src/components/AddressField.tsx (hunk 8 above)
Comment (lucachaco) [reply src/components/AddressField.tsx:8]: true, on it

Comment (lucachaco) [reply src/components/AddressField.tsx]: Done

**Review by ilanolkies — APPROVED** [review]
Great job!

**Review by chescalante — APPROVED** [review]

**Pull request merged**